### DISCUSSION TITLE
DIMER for 7" official raspberry screen

### DISCUSSION THREAD

**Discussion by Rasb74** [rern/rAudio#699] [Feature Requests]
Hi, 
Following and supporting your famous dev form the beginning of the story, I also regularly update my components.
Now using a 7" official screen for test, everything is ok, instead of the high brightness of the screen. 
It would be interesting to propose a dimmer, as you already propose a ON/OFF switch, I guess it will be a nice feature. 
Cheers

**Comment by rern** [comment]
Try this for me:
```sh
# existing value
cat /sys/class/backlight/rpi_backlight/brightness
# set at 50%
echo 128 > /sys/class/backlight/rpi_backlight/brightness
# set at 100%
echo 255 > /sys/class/backlight/rpi_backlight/brightness
```

**Comment by Rasb74** [comment]
Hello !

Thanks for your reply. It works perfectly, 

Would be interesting to implement gradient brightness values like this :

OFF
20
40
60
100
255
ON

If someone else interesting, may a live ambient light sensor can be interesting in the future…

One more time congrats for your great job ! 

Cheers

Pascal

**Reply by rern** [reply]
Can you recall output of the 1st command?
It tells the file exists or not before setting values. I'm trying to find an indicator of the connected display to show an option in the Settings.
Do you have to configure something to enable the display?
If none, please try:
- Disconnect the display > Power on
- Does `/sys/class/backlight/rpi_backlight/brightness` still exist?

**Comment by Rasb74** [comment]
Sorry what do you mean by recall output . Sorry my coding skills are limited !
But I don’t have any message indicating a non existing file or folder. 
Nothing as to be configured to enable the display. It’s a SDI connection, only I’ve to select “Browser on RPi” in R-Audio features.

**Reply by rern** [reply]
All right. Let's start over again.
1. Reboot
2. command: `[[ -e /sys/class/backlight/rpi_backlight/brightness ]] && echo YES || echo NO` - Note the output, YES or NO.

If the output is NO, that's enough.

If YES, continue:

3. Power off
4. Disconnect jumper and ribbon cables of 7" display from RPi
5. Power on
6. command: `[[ -e /sys/class/backlight/rpi_backlight/brightness ]] && echo YES || echo NO` - Note the output, YES or NO.
7. Power off
8. Reconnect jumper and ribbon cables of 7" display to RPi
9. Power on
10. Done

**Comment by Rasb74** [comment]
Hi Rern,

I’ve made the test the results :
> All right. Let's start over again.
> 
> Reboot
> command: [[ -e /sys/class/backlight/rpi_backlight/brightness ]] && echo YES || echo NO - Note the output, YES or NO.
YES

> Power off
> Disconnect jumper and ribbon cables of 7" display from RPi
> Power on
> command: [[ -e /sys/class/backlight/rpi_backlight/brightness ]] && echo YES || echo NO - Note the output, YES or NO.
NO


Additionnaly, I noticed the system reactivity looks a bit slower when the screen is used. 
I’m using a RPI4B 8Gb.

cheers

Pascal

**Comment by rern** [comment]
Try today's update > Browser on RPi > Brightness setting

**Comment by Rasb74** [comment]
hi fern,
feature doesn’t appear even after reboot
cheers

**Reply by rern** [reply]
Hi Pascal,

Try:
- Reboot
- Note the output of this command:
  - `cat /sys/class/backlight/rpi_backlight/brightness`
- Then:
  - `echo 255 > /sys/class/backlight/rpi_backlight/brightness`
- Try the setting of Browser on RPi again

**Comment by Rasb74** [comment]
Hello Rern,
> cat /sys/class/backlight/rpi_backlight/brightness
> Then:
255

Still no setting on browser on RPI.

**Reply by rern** [reply]
Hi Pascal,

Try:
```sh
wget https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/assets/js/features.js -O /srv/http/assets/js/features.js
```

**Comment by Rasb74** [comment]
Hello,

Still nothing in the features browser on the RPI

**Reply by rern** [reply]
Does today's update show the brightness setting?

**Comment by Rasb74** [comment]
hello Re’n;
yes the brightness setting now appear, but no action. 
anyway, now we can see it, next step is to make it working.

cheers and a merry christmas !

**Reply by rern** [reply]
Run this command `chmod 666 /sys/class/backlight/rpi_backlight/brightness` and try adjusting the brightness again.

**Comment by Rasb74** [comment]
Hello Rern,

Works perfectly ! 
One more time you’re great !!!

Some comment : 
When using the on screen display (browser on RPI), I encounter a big latency between the playlist/main screen/Librairy panels on the web interface.
If I unconnect the screen, all of this is very reactive.
Can you check this matter?

Cheers

Pascal

**Comment by ghost** [comment]
The slider is visible in the (browser on RPI) menu. Setting the slider to a new (lower) position doesn't lower the brightness. You have to reboot the device before the actual brightness of the background light goes to the desired level. Is this the way it is supposed to work?

**Comment by Leobo71** [comment]
My 7 "LCD is connected with a Y cable.   When I put that in  `chmod 666 /sys/class/backlight/rpi_backlight/brightness`.  Is working. After reboot it does not work. When the RPI is switched off, the power LED on the raspi light up. Is it possible to switch off the RPI completely? Thanks
